labgrid/remote/client.py

```diff
@@ -790,19 +786,34 @@ def _get_ip(self, place):
         return newest[0]
 
     def ssh(self):
+        from ..resource import NetworkService
         place = self.get_acquired_place()
         ip = self._get_ip(place)
         if not ip:
             return
-        args = [
+        target = self._get_target(place)
+        env = os.environ.copy()
+        try:
+            resource = target.get_resource(NetworkService)
+            username = resource.username
+            # use sshpass if we have a password
+            if resource.password:
+                env['SSHPASS'] = resource.password
+            sshpass = ['sshpass', '-e'] if resource.password else []
```

Comment:
You need to check if there the port is 22 or not. Otherwise it won't work with the ForwardedNetworkService.
```
port = resource.port or 22
```